src/components/Asset/AssetActions/Trade/Swap.tsx

```diff
@@ -7,44 +7,49 @@ import { FormikContextType, useFormikContext } from 'formik'
 import Output from './Output'
 import Slippage from './Slippage'
 import PriceImpact from './PriceImpact'
-
 import Decimal from 'decimal.js'
 import { useAsset } from '@context/Asset'
+import { useWeb3 } from '@context/Web3'
 import { FormTradeData, TradeItem } from './_types'
-import { Asset } from '@oceanprotocol/lib'
+import { LoggerInstance, Pool } from '@oceanprotocol/lib'
 import { AssetExtended } from 'src/@types/AssetExtended'
+import { usePool } from '@context/Pool'
 
 Decimal.set({ toExpNeg: -18, precision: 18, rounding: 1 })
 
 export default function Swap({
-  asset,
+  assetExtended,
```

Comment:
the prop name can just stay as `asset` which is also consistent to all other components when using this prop